### ISSUE TITLE
useIsFetching causes react (16.13.x) to throw warnings (unable to update while rendering)

### ISSUE THREAD

**Issue by mustafawm** [tannerlinsley/react-query#334]
Upon the release of react to 16.13.x , `useIsFetching` started throwing this error: "Warning: Cannot update a component X while rendering a different component.. "
related [18178](https://github.com/facebook/react/issues/18178)

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/13101565/77882692-b02d4980-728b-11ea-9114-b1d1ffbefc95.gif)
[codesandbox](https://codesandbox.io/s/nostalgic-shaw-v39d6) (_warning in console_)

It seems to be an issue with `useBaseQuery/useIsFetching` 

`useBaseQuery` updates the `cache.isFetching` which triggers `useIsFetching` hook which causes components using it to re-render

```js
function CeoComponent() {
  // 👇 triggers useIsFetching before it (<CeoComponent />) has finished rendering
  const { data, status } = useQuery(["ceo", 2], fetchCeo); rendering

  return (
    <div className="">
      <h2>dog 2</h2>
      {status === "success" && data ? (
        <img alt="dog ceo" src={data.message} />
      ) : null}
    </div>
  );
}

function GlobalLoadingIndicator() {
  const isFetching = useIsFetching();

  return (
    <div className="">
      <div>global isFetching: {isFetching ? "true" : "false"}</div>
    </div>
  );
}
```

**Comment by jprosevear** [comment]
I encountered something similar but it turned out to be the react-query devtools - can you reproduce with those running?

**Comment by mustafawm** [comment]
I commented out react-query devtools in the codesandbox provided and the warning is still there.

I think `useBaseQuery` needs to update `isFetching` in a `useEffect` so it happens after the rendering is done
_I couldn't get the source code/examples to run locally to be able to debug/test myself_

**Comment by jprosevear** [comment]
@mustafawm #289 ?

**Comment by mustafawm** [comment]
@jprosevear nope.

**Comment by tannerlinsley** [comment]
This is fixed in 1.2.2. Thanks!

**Comment by rkulinski** [comment]
@tannerlinsley 
For this mix:
```
    "react-query": "1.5.10",
    "react-query-devtools": "1.2.0",
    "react": "16.13.1",
```

That's still an issue.
